In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [8]:
import scipy.stats as stats

In [2]:
df = pd.read_csv('ab_data.csv')

In [3]:
df.head(5)

user_id                   timestamp      group landing_page  converted
0   851104  2017-01-21 22:11:48.556739    control     old_page          0
1   804228  2017-01-12 08:01:45.159739    control     old_page          0
2   661590  2017-01-11 16:55:06.154213  treatment     new_page          0
3   853541  2017-01-08 18:28:03.143765  treatment     new_page          0
4   864975  2017-01-21 01:52:26.210827    control     old_page          1

## Hypothesis Testing Setup

In this experiment a new landing page for website selling milk tea (for example) is being tested to see if it generates a greater proportion of visitors that buy 1 or more milk tea products.

The landing page generates either the new or the old landing page randomly for a visitor, and the for the same visitor, we track whether a milk tea product is bought during their session that they explore the milk tea website.

The null hypothesis is that the proportions of visitors who land on the new landing page will be the same as the proportion of visitors landing on the old landing page.

The alternative hypothesis is that the vistors with the new landing page will have a greater proportion that buy milk tea products during their session exploring the website.

H0: pT - pC = 0
HA: pT - pC > 0

And will we will set the necessary p-value cutoff for significance at p = 0.05

In [5]:
pd.crosstab(df.group, df.converted, margins = True)

converted       0      1     All
group                           
control    129479  17723  147202
treatment  129762  17514  147276
All        259241  35237  294478

In [7]:
pd.crosstab(df.group, df.converted).apply(lambda r: r/r.sum(), axis=1)

converted         0         1
group                        
control    0.879601  0.120399
treatment  0.881080  0.118920

In [11]:
treatment = df.loc[df['group'] == 'treatment', 'converted']
control = df.loc[df['group'] == 'control', 'converted']

In [23]:
tval, pval = stats.ttest_ind(control, treatment, equal_var=True)

In [24]:
tval

1.2369207675839395

In [25]:
pval

0.21611748562490837

## Hypothesis Testing Analysis

Given that the p-value from this t-test is 0.216 and is greater than 0.05, we fail to reject the null hypothesis and conclude there is not a significant difference in purchases between vistors who viewed the old and new landing pages.